storage/src/main/java/fk/prof/storage/D42AsyncStorage.java

```diff
@@ -36,15 +36,19 @@ public D42AsyncStorage(AmazonS3 client, String bucket, ExecutorService executorS
     }
 
     @Override
-    public void storeAsync(String path, InputStream content) {
+    public void storeAsync(String path, InputStream content, long length) {
         CompletableFuture.runAsync(() -> {
             try {
-                client.putObject(bucket, path, content, new ObjectMetadata());
+                ObjectMetadata meta = new ObjectMetadata();
+                meta.setContentLength(length);
+                client.putObject(bucket, path, content, meta);
+                // TODO expose metric for size written
```

Comment:
vertx has a metricsservice for emitting metrics. How do you intend to emit metrics here? (since this is vertx agnostic)